src/prefect/_internal/concurrency/from_async.py

```diff
@@ -45,10 +43,8 @@ def call_soon_in_worker_thread(
     Returns a supervisor.
     """
     runtime = get_runtime_thread()
-    supervisor = AsyncSupervisor()
-    with set_supervisor(supervisor):
-        future = runtime.submit_to_worker_thread(__fn, *args, **kwargs)
-    supervisor.set_future(future)
+    supervisor = AsyncSupervisor(runtime.submit_to_worker_thread)
+    supervisor.submit(__fn, *args, **kwargs)
```

Comment:
This diff shows how much easier it is to use the supervisor correctly

Reply:
Very nice!